climada/util/alpha_shape.py

```diff
@@ -0,0 +1,99 @@
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Compute the concave hull of a set of points.
+"""
+
+import numpy as np
+
+from shapely.ops import cascaded_union, polygonize
+from scipy.spatial import Delaunay
+import math
+import shapely.geometry as geometry
+import pylab as pl
+
+def alpha_shape(points, alpha):
+    """
+    Compute the alpha shape (concave hull) of a set
+    of points.
+    Source: https://thehumangeo.wordpress.com/2014/05/12/drawing-boundaries-in-python/
+    @param points: Iterable container of points.
+    @param alpha: alpha value to influence the
+        gooeyness of the border. Smaller numbers
+        don't fall inward as much as larger numbers.
+        Too large, and you lose everything!
+    """
+    if len(points) < 4:
+        # When you have a triangle, there is no sense
+        # in computing an alpha shape.
+        return geometry.MultiPoint(list(points)).convex_hull
+
+    def add_edge(edges, edge_points, coords, i, j):
+        """
+        Add a line between the i-th and j-th points,
+        if not in the list already
+        """
+        if (i, j) in edges or (j, i) in edges:
+            # already added
+            return
+        edges.add( (i, j) )
+        edge_points.append(coords[ [i, j] ])
+    coords = np.array([point.coords[0]
+                       for point in points])
+    tri = Delaunay(coords)
+    edges = set()
+    edge_points = []
+    # loop over triangles:
+    # ia, ib, ic = indices of corner points of the
+    # triangle
+    for ia, ib, ic in tri.vertices:
+        pa = coords[ia]
+        pb = coords[ib]
+        pc = coords[ic]
+        # Lengths of sides of triangle
+        a = math.sqrt((pa[0]-pb[0])**2 + (pa[1]-pb[1])**2)
+        b = math.sqrt((pb[0]-pc[0])**2 + (pb[1]-pc[1])**2)
+        c = math.sqrt((pc[0]-pa[0])**2 + (pc[1]-pa[1])**2)
+        # Semiperimeter of triangle
+        s = (a + b + c)/2.0
+        # Area of triangle by Heron's formula
+        area = math.sqrt(s*(s-a)*(s-b)*(s-c))
+        if area > 0:
+            circum_r = a*b*c/(4.0*area)
+            # Here's the radius filter.
+            #print circum_r
+            if circum_r < 1.0/alpha:
+                add_edge(edges, edge_points, coords, ia, ib)
+                add_edge(edges, edge_points, coords, ib, ic)
+                add_edge(edges, edge_points, coords, ic, ia)
+    m = geometry.MultiLineString(edge_points)
+    triangles = list(polygonize(m))
+    return cascaded_union(triangles), edge_points
+
+def plot_polygon(polygon):
+    fig = pl.figure(figsize=(10,10))
+    ax = fig.add_subplot(111)
+    margin = .3
+    x_min, y_min, x_max, y_max = polygon.bounds
+    ax.set_xlim([x_min-margin, x_max+margin])
+    ax.set_ylim([y_min-margin, y_max+margin])
+    for geom in polygon.geoms:
+        xs, ys = geom.exterior.xy
+        ax.fill(xs, ys, fc='#999999', ec='#000000', fill=True)
+
+    return fig
```

Comment:
This was  blindly copied from another source  and is not in the same style  as other climada plotting functions. Is this  really needed?  If yes, please adapt. See other plotting methods  in the util functions.

Reply:
I'll update when we decide to keep it =)

Reply:
I've removed the alpha_shape and the hull_bruned_area method in commit 13a2c8c as calculating area burned is not used and the module is complex already